codegeex.el

```diff
@@ -31,56 +32,48 @@
 (require 'url)
 (require 'json)
 (require 'uuidgen)
+(require 'codegeex-completion)
+(require 'codegeex-overlay)
 
-(defvar codegeex-endpoint "https://tianqi.aminer.cn/api/v2/" "the endpoint of CodeGeeX API")
-(defvar codegeex-apikey "68cf004321e94b47a91c2e45a8109852" "API key obtained from CodeGeeX website")
-(defvar codegeex-apisecret "e82b86a16f9d471ab215f653060310e3" "API secret obtained from CodeGeeX website")
-(defvar codegeex-temperature 0.2 "temperature for completion by CodeGeeX")
-(defvar codegeex-top_p 0.95 "top_p for completion by CodeGeeX")
-(defvar codegeex-top_k 0 "top_k for completion by CodeGeeX")
```

Comment:
Could be subject to an other pull request
These could be defcustom. And currently you put your own api key and api secret. Which is not ideal. 
Maybe we can create a login method like the vscode plugin. Retrieving the api with it. 
But I have no idea how the api is working.

Some explanation of top_p and top_k or temperature could be added as well.

Reply:
In fact, they are not my private api key and api secret, they are copied from vscode extension.

Based on my previous test, this extension uses some of non-open API of codegeex(e.g. fill in middle API), which can be accessed only if using this key.

Maybe @Stanislas0 could provide more information about it.

Reply:
The API is kind of shady it seems. I guess some clarification would be needed indeed.

> copied from vscode extension
You mean [this](https://github.com/CodeGeeX/codegeex-vscode-extension/tree/main/src) one? I don't even know if it's the source code of the current extension shown in the marketplace.

Reply:
We can clearly see that the completion proposed with the API for this package is not the same than the one from the vscode plugin.

![image](https://github.com/hzhangxyz/codegeex.el/assets/34922853/9512556a-10c0-429b-8c14-d562df53f14e)
Compared to:
![image](https://github.com/hzhangxyz/codegeex.el/assets/34922853/bb2daeaa-6a3a-4e1c-a2d2-4abc636ad517)

Clearly there is a new version of the API or something need to be changed in the Call itself.

I'll create an issue on the CodeGeeX2 github project since you get no news from @Stanislas0 in the issue #1.

Reply:
Well, that change in the completion proposed was my fault, sorry for that dumb mistake (fixed in last commits). 

I've checked the files of the vscode extension, I didn't know the whole code was available. 
Indeed everything could be retrieved from it. But I hope a better documentation would be available.